root/usr/libexec/nethserver/openvpn-status3

```diff
@@ -40,12 +44,45 @@ while (defined(my $msg = <TSOCK>)) {
             # Header, Common Name Real Address    Virtual Address Virtual IPv6 Address    Bytes Received  Bytes Sent  Connected Since Connected Since (time_t)    Username    Client ID   Peer ID
             my ($header, $cn, $real_address, $virtual_ipv4, $virtual_ipv6, $bytes_received, $bytes_sent, $since, $username, $client_id, $peer_id) = split('\t', $msg);
             next if ($cn eq 'UNDEF');
+
+            # rerieve the parameters to find information in conntrack
+            #udp      17 172 src=xxx.xxx.xxx.xxx dst=xxx.xxx.xxx.xxx sport=57045 dport=1194 packets=169 bytes=32971 
+            #src=xxx.xxx.xxx.xxx dst=xxx.xxx.xxx.xxx sport=1194 dport=57045 packets=68 bytes=14663 [ASSURED] mark=0 delta-time=27 use=1
+            my ($ipConntrack,$portConntrack) = split(':',$real_address);
+            my ($provider,$RedInterface);
+            my $protocol = ($db->get_prop('openvpn@host-to-net', 'UDPPort') ne '') ? 'udp' : 'tcp';
+
+            open(FH, "/usr/sbin/conntrack -L -p $protocol --src $ipConntrack --sport $portConntrack|");
+            my @fields;
+            while (<FH>) {
+                @fields = split('\s');
+                foreach (@fields) {
+                    # Compare and Find the good IP used in red NIC
+                    next if ($_ !~ /^src=/);
+                    next if ($_ =~ /$ipConntrack/);
+                    $_ =~ s/^src=//;
+                    $RedInterface = $_ ;
+                }
+            }
+            close(FH);
```

Comment:
I've tested it and it seems to me a little over complicated.
We should be able to identify a single connection using source, protocol, source port, destination port
So we could get rid of the double loop and just analyze the first line returned from `conntrack` command.

Something like:
```suggestion
            my $port = $db->get_prop('openvpn@host-to-net', 'Port') || '1194';

            my $out = `/usr/sbin/conntrack -L -p $protocol --src $ipConntrack --sport $portConntrack --dport $port 2>/dev/null`;
            $out =~ m/\sdst=(\d+\.\d+\.\d+\.\d+)\s/;
            $RedInterface = $1;
```

Does it make sense to you?

Reply:
magic of regex yes it makes sense, not tested, will do it later

Reply:
better thanks